src/libcore/iter.rs

```diff
@@ -693,7 +717,10 @@ pub trait Iterator {
             if predicate(x) {
                 return Some(i);
             }
-            i += 1;
+            i = match i.checked_add(1) {
+                Some(ii) => ii,
+                None => panic!("overflow while getting the position of an element in an iterator"),
+            }
```

Comment:
This may also have performance considerations like in `count` (and is basically never expected to trip)